library/x509_crt.c

```diff
@@ -2982,6 +2984,124 @@ static int x509_crt_verify_chain(
     }
 }
 
+/*
+ * parse ipv4 address from canonical string form into bytes.
+ * return 0 if success, -1 otherwise
+ */
+MBEDTLS_STATIC_TESTABLE int x509_parse_ipv4( const char *h, unsigned char *addr )
+{
+    int i;
+    const char *strt = h;
+    char *endp;
+    unsigned long v;
+    for( i = 0; i < 4; i++ )
+    {
+        v = strtoul( strt, &endp, 10 );
+        if( endp == strt || v > 255 )
+        {
+            return( -1 );
+        }
+
+        if( i < 3 && *endp != '.' )
+        {
+            return( -1 );
+        }
+        *( addr + i ) = ( unsigned char )( v & 0xFFU );
+        strt = endp + 1;
+    }
+
+    if( *endp != '\0' ) // trailing data
+    {
+        return( -1 );
+    }
+
+    return( 0 );
+}
+
+/*
+ * parse ipv6 address from canonical string form into bytes.
+ * return 0 if success, -1 otherwise
+ */
+MBEDTLS_STATIC_TESTABLE int x509_parse_ipv6( const char *h, size_t hlen, unsigned char *addr )
```

Comment:
The CI (more precisely, `tests/scripts/check-names.sh`) reminds me that, since this function will be technically public in some configurations, its name needs to be properly prefixed with `mbedtls_` (same for the ipv4 function obviously).

Reply:
:checkered_flag: